src/errors.js

```diff
@@ -0,0 +1,86 @@
+let uuid = require('uuid');
+
+const ERROR_CODES = {
+  MalformedPayload:         400,  // Only for JSON.parse() errors
+  InputValidationError:     400,  // Only for JSON schema errors
+  AuthorizationFailed:      401,  // Only if authentication failed
+  InsufficientScopes:       403,  // Only if request had insufficient scopes
+  ResourceNotFound:         404,  // If the resource wasn't found
+  RequestConflict:          409,  // If the request conflicts with server state
+  ResourceExpired:          410,  // If the resource expired over time
+  InputTooLarge:            413,  // Only if the payload is too big
+  InternalServerError:      500,  // Only for internal errors
+};
+
+// Export ERROR_CODES
+exports.ERROR_CODES = ERROR_CODES;
+
+/**
+ * Middleware that adds `res.reportError(code, message, details)` and
+ * `res.reportInternalError(error)`.
+ *
+ * The `method` is the name of the API method, `errorCodes` is a mapping from
+ * allowed error codes to HTTP status codes, and `raven` is an instance of
+ * `raven.Client` from the `raven` npm module.
+ */
+let BuildReportErrorMethod = (method, errorCodes, ravent = null) => {
+  return (req, res, next) => {
+    res.reportError = (code, message, details = {}) => {
+      let status = errorCodes[code];
+      if (status === undefined || typeof(message) !== 'string') {
```

Comment:
If message is missing it will be included in the 500 error:
```js
 message = 'Internal error, unknown error code: ' + code + '\n' +
              (message || 'Missing message!');
```